editor/editor_inspector.cpp

```diff
@@ -1954,7 +1962,8 @@ void EditorInspector::update_tree() {
 					ep->set_checked(checked);
 					ep->set_keying(keying);
 
-					ep->set_read_only(read_only);
+					bool property_read_only = (p.usage & PROPERTY_USAGE_READ_ONLY) || read_only;
```

Comment:
You can move this out of the `for (Ref<EditorInspectorPlugin> &ped : valid_plugins)` loop, similar to how it's done for `checked` and `checkable`